chapters/statemachines.tex

```diff
@@ -428,35 +428,18 @@ \subsection{Merging Variable Definitions}\label{merging-variable-definitions}
 \subsection{Merging Connections to Multiple Outputs}\label{merging-connections-to-multiple-outputs}
 
 \begin{nonnormative}
-Since instances of blocks can be used as states of a state
-machine it is natural to extend the connection semantics of Modelica to
-allow several outputs to be connected to one input.
+The causal connection semantics of Modelica for non-state machines are generalized to states of state machines, using the fact that only one state is active at a time.
 \end{nonnormative}
 
-It is possible to connect several outputs to an input if all the outputs
-come from states of the same state machine. In such cases, we get the
-following constraint equations:
+It is possible to connect outputs each coming from different states of state machines together - and connect this with other causal connectors.
+These outputs are combined seen as one source of the signal, and give the following constraint equations,
 \begin{lstlisting}[language=modelica,escapechar=!]
 u!\textsubscript{1}! = u!\textsubscript{2}! = ... = y!\textsubscript{1}! = y!\textsubscript{2}! = ...
 \end{lstlisting}
-with u\textsubscript{i} inputs and y\textsubscript{i} outputs. The
-semantics is defined as follows. Introduce a variable v representing the
-signal flow and rewrite the equation above as a set of equations for
-u\textsubscript{i} and a set of assignment equations for v:
+with \lstinline!y!\textsubscript{i} being outputs from different states of the state-machine and \lstinline!u!\textsubscript{i} being other causal variables.
+The semantics is defined similarly to \cref{merging-variable-definitions}:
 \begin{lstlisting}[language=modelica,escapechar=!]
-v := if activeState(state!\textsubscript{1}!) then y!\textsubscript{1}! else last(v);
-v := if activeState(state!\textsubscript{2}!) then y!\textsubscript{2}! else last(v);
-...
-u!\textsubscript{1}! = v
-u!\textsubscript{2}! = v
-...
-\end{lstlisting}
-
-The merge of the definitions of v is then made according to \cref{merging-variable-definitions}:
-Merging Variable Definitions. The result is after
-simplification:
-\begin{lstlisting}[language=modelica,escapechar=!]
-v := if activeState(state!\textsubscript{1}!) then y!\textsubscript{1}! elseif activeState(state!\textsubscript{2}!) then y!\textsubscript{2}! elseif ... else last(v);
+v = if activeState(state!\textsubscript{1}!) then y!\textsubscript{1}! elseif activeState(state!\textsubscript{2}!) then y!\textsubscript{2}! elseif ... else last(v);
```

Comment:
Not new, but we should take the opportunity to break this long line:
```suggestion
v = if activeState(state!\textsubscript{1}!) then
      y!\textsubscript{1}!
    elseif activeState(state!\textsubscript{2}!) then
      y!\textsubscript{2}!
    elseif $\ldots$
    else
      last(v);
```

(I know there's a different way to arrange an if-expression like this, using only one line per branch, but I'm not aware of us using that style anywhere else.)